services/sagemaker/pkg/resource/model/custom_update_api.go

```diff
@@ -0,0 +1,29 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package model
+
+import (
+	"context"
+	ackcompare "github.com/aws/aws-controllers-k8s/pkg/compare"
+)
+
+// TrainingJob has no update op (as of aws_sdk_go_v1.34.32)
+func (rm *resourceManager) customUpdateModel(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	diffReporter *ackcompare.Reporter,
+) (*resource, error) {
+	return latest, nil
```

Comment:
s/TrainingJob/Model/

If the Model doesn't have an update op, why aren't we keeping the default behaviour of returning a NotImplemented error into the reconciler loop?

Reply:
@surajkota @RedbackThomson Jay's point makes sense. This change was in keeping with the current operator experience. Opinions ?

Reply:
Is there a clean way to report back to the user that this operation is not supported?

Reply:
Unfortunately, I think the "cleanest" way is the default behaviour of returning a NotImplemented error into the reconciler loop and relying on the error logging that occurs already.

Reply:
Should we set a condition for such cases then? maybe OutOfSync? because if a user updates this resource and it cannot reach the desired state

Reply:
@surajkota yeah, we should probably just consider NotImplemented to be a Terminal Condition...